xarray/core/indexes.py

```diff
@@ -1925,6 +1946,36 @@ def default_indexes(
     return indexes
 
 
+def _wrap_index_equals(
+    index: Index,
+) -> Callable[[Index, frozenset[Hashable]], bool]:
+    # TODO: remove this Index.equals() wrapper (backward compatibility)
+
+    sig = inspect.signature(index.equals)
+
+    if len(sig.parameters) == 1:
+        index_cls_name = type(index).__module__ + "." + type(index).__qualname__
+        emit_user_level_warning(
+            f"the signature ``{index_cls_name}.equals(self, other)`` is deprecated. "
+            f"Please update it to "
+            f"``{index_cls_name}.equals(self, other, *, exclude_dims=None)`` "
+            "or kindly ask the maintainers doing it. "
```

Comment:
```suggestion
            "or kindly ask the maintainers of ``{index_cls_name}`` to do it. "
```

otherwise they might come to us ;)